lte/gateway/python/magma/subscriberdb/protocols/s6a_proxy_servicer.py

```diff
@@ -109,32 +114,24 @@ def UpdateLocation(self, request, context):
         apn.ambr.max_bandwidth_dl = profile.max_dl_bit_rate
         apn.pdn = s6a_proxy_pb2.UpdateLocationAnswer.APNConfiguration.IPV4
 
-        # TODO - Add APN config through CLI
-        # ims apn
-        apn_ims = ula.apn.add()
-        apn_ims.context_id = 1
-        apn_ims.service_selection = 'ims'
-        apn_ims.qos_profile.class_id = 5
-        apn_ims.qos_profile.priority_level = 15
-        apn_ims.qos_profile.preemption_capability = 1
-        apn_ims.qos_profile.preemption_vulnerability = 0
-
-        apn_ims.ambr.max_bandwidth_ul = profile.max_ul_bit_rate
-        apn_ims.ambr.max_bandwidth_dl = profile.max_dl_bit_rate
-        apn_ims.pdn = s6a_proxy_pb2.UpdateLocationAnswer.APNConfiguration.IPV4
-
-        # internet apn
-        apn_internet = ula.apn.add()
-        apn_internet.context_id = 2
-        apn_internet.service_selection = 'internet'
-        apn_internet.qos_profile.class_id = 1
-        apn_internet.qos_profile.priority_level = 5
-        apn_internet.qos_profile.preemption_capability = 1
-        apn_internet.qos_profile.preemption_vulnerability = 0
-
-        apn_internet.ambr.max_bandwidth_ul = profile.max_ul_bit_rate
-        apn_internet.ambr.max_bandwidth_dl = profile.max_dl_bit_rate
-        apn_internet.pdn = s6a_proxy_pb2.UpdateLocationAnswer.APNConfiguration.IPV4
-
-
+        num_sec_apn = len(sub_data.non_3gpp.apn_config)
+        for idx in range(num_sec_apn):
+            sec_apn = ula.apn.add()
+            # Context id 0 is assigned to oai.ipv4 apn. So start from 1
+            sec_apn.context_id = idx + 1
+            sec_apn.service_selection = sub_data.non_3gpp.apn_config[
+                idx
+            ].service_selection
+            sec_apn.qos_profile.class_id = sub_data.non_3gpp.apn_config[
+                idx
+            ].qos_profile.class_id
+            sec_apn.qos_profile.priority_level = 15
+            sec_apn.qos_profile.preemption_capability = 1
+            sec_apn.qos_profile.preemption_vulnerability = 0
+
+            sec_apn.ambr.max_bandwidth_ul = profile.max_ul_bit_rate
+            sec_apn.ambr.max_bandwidth_dl = profile.max_dl_bit_rate
```

Comment:
why don't we configure these through the CLI?

Reply:
Since these are the default values(as per the spec) there is no need to configure from CLI/script

Reply:
Default values regardless of the QoS class? Could you point me to the relevant specification?

Reply:
spec 29.272 sec 7.3.40 and 29.212 sec 5.3.45

Reply:
The referenced documents do not provide default values, but specify range of values. These specify the QoS vector for the default bearer of the APN. As such, they are configurable things.